libtelemetry/src/main/java/com/mapbox/services/android/telemetry/PermissionCheckRunnable.java

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.services.android.telemetry;
+
+import android.content.Context;
+import android.os.Handler;
+
+import com.mapbox.services.android.core.permissions.PermissionsManager;
+
+class PermissionCheckRunnable implements Runnable {
+  private final Context context;
+  private final Handler handler = new Handler();
+  private final ExponentialBackoff counter = new ExponentialBackoff();
+  private MapboxTelemetry mapboxTelemetry;
+
+  PermissionCheckRunnable(Context context, MapboxTelemetry mapboxTelemetry) {
+    this.context = context;
+    this.mapboxTelemetry = mapboxTelemetry;
+  }
+
+  @Override
+  public void run() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      mapboxTelemetry.optIn();
```

Comment:
I guess this won't work properly because `mapboxTelemetry.optIn()` 👉 `optLocationIn()` 👉 `checkLocationPermission()` 👉 `permissionBackoff()` which will create a new instance of `PermissionCheckRunnable` so the intervals will be reset (loosing the exponential backoff behavior).
What about creating a lazy initialization method for `PermissionCheckRunnable` creation?